regolith/helpers/u_milestonehelper.py

```diff
@@ -110,161 +102,173 @@ def construct_global_ctx(self):
+            if not target_prum:
+                pra = fragment_retrieval(self.gtx["projecta"], ["_id"],
+                                         rc.projectum_id)
+                print("Projectum not found. Projecta with similar names: ")
+                for i in range(len(pra)):
+                    print(f"{pra[i].get('_id')}")
+                print("Please rerun the helper specifying the complete ID.\n"
+                      "If your prum id looks correct, check that this id is in the collection "
+                      "in the database that regolith is looking in (i.e., {rc.database}).\n"
+                      "If this is the case, rerun with --database set to the database where the item is located.")
+                return
+            milestones = deepcopy(target_prum.get('milestones'))
+            all_milestones = []
+            for item in milestones:
+                item['identifier'] = 'milestones'
+            all_milestones.extend(milestones)
+            for i in all_milestones:
+                i['due_date'] = get_due_date(i)
+            all_milestones.sort(key=lambda x: x['due_date'], reverse=False)
+            index_list = list(range(2, (len(all_milestones) + 2)))
+            mil = {}
+            if not rc.due_date or not rc.name or not rc.objective:
+                raise RuntimeError(
+                    "name, objective, and due date are required for a new milestone")
+            mil.update({'due_date': rc.due_date})
+            mil['due_date'] = get_due_date(mil)
+            mil.update({'objective': rc.objective, 'name': rc.name, 'uuid': get_uuid()})
+            if rc.audience:
+                mil.update({'audience': rc.audience})
+            else:
+                mil.update({'audience': ['lead', 'pi', 'group_members']})
+            if rc.status:
+                mil.update({'status': rc.status})
+            else:
+                mil.update({'status': 'proposed'})
+            if rc.notes:
+                mil.update({'notes': rc.notes})
+            if rc.type:
+                if rc.type in MILESTONE_TYPES:
                     mil.update({'type': rc.type})
                 else:
-                    mil.update({'type': 'meeting'})
-                if rc.status:
-                    mil.update({'status': rc.status})
-                else:
-                    mil.update({'status': 'proposed'})
-                if rc.notes:
-                    mil.update({'notes': rc.notes})
-                mil.update({'identifier': "milestones"})
-                new_mil.append(mil)
-                pdoc = {'milestones': mil}
-            if idx > 1:
-                doc = deepcopy(all_milestones[idx - 2])
-                identifier = doc['identifier']
-                if not doc.get(
-                        'type') and not rc.type and identifier == 'milestones':
-                    raise RuntimeError(
-                        "ERROR: This milestone does not have a type set and this is required. "
```

Comment:
@sbillinge I removed the RuntimeError that was previously on lines 214-215. Are there any cases in which existing milestones would not have a meeting type?  I will put it back in the code for now because there are milestones on the schema without `type` specified.

Reply:
It should be there.  We want `type` to be required and selected from an `eallowed` list

Reply:
Done.